docs/developers/sbt.md

```diff
@@ -0,0 +1,277 @@
+```
+
+Similarly, you can build the Flink 1.15 client assembly jar using the following commands:
+
+```shell
+$ ./build/sbt
+> project celeborn-client-flink-1_15-shaded
+> assembly
+
+$ # Or, you can use sbt directly with the `-Pflink-1.15` profile:
+$ ./build/sbt -Pflink-1.15 celeborn-client-flink-1_15-shaded/assembly
+```
+
+By executing these commands, you will create assembly jar files for the respective Spark and Flink client modules. The assembly jar bundles all the dependencies, allowing the client module to be used independently with all required dependencies included.
+
+## Building submodules individually
+
+For instance, you can build the Celeborn Client module using:
+
+```
+$ # sbt
+$ build/sbt
+> project celeborn-client
+> package
+
+$ # Or, you can build the celeborn-client module with sbt directly using:
+$ build/sbt celeborn-client/package
+```
+
+## Testing with SBT
+
+To run all tests for the Celeborn project, you can use the following command:
+
+```shell
+./build/sbt test
+```
+
+Running tests for specific versions of Spark/Flink client.
+
+For example, to run the test cases for the Spark 3.3 client, use the following command:
+
+```shell
+$ ./build/sbt -Pspark-3.3 test
+
+$ # only run spark client related modules tests
+$ ./build/sbt -Pspark-3.3 celeborn-spark-group/test
+```
+
+Similarly, to run the test cases for the Flink 1.15 client, use the following command:
+
+```shell
+$ ./build/sbt -Pflink-1.15 test
+
+$ # only run flink client related modules tests
+$ ./build/sbt -Pflink-1.15 celeborn-flink-group/test
+```
+
+## Running Individual Tests
+
+When developing locally, it’s often convenient to run a single test or a few tests, rather than running the entire test suite.
+
+The fastest way to run individual tests is to use the sbt console. It’s fastest to keep a sbt console open, and use it to re-run tests as necessary. For example, to run all of the tests in a particular project, e.g., client:
+
+```
+$ build/sbt
+> project celeborn-client
+> test
+```
+You can run a single test suite using the `testOnly` command. For example, to run the `ShuffleClientSuiteJ`:
+
+```
+> testOnly org.apache.celeborn.client.ShuffleClientSuiteJ
+```
+The `testOnly` command accepts wildcards; e.g., you can also run the `ShuffleClientSuiteJ` with:
+
+```
+> testOnly *ShuffleClientSuiteJ
+```
+Or you could run all of the tests in the `client` package:
+
+```
+> testOnly org.apache.celeborn.client.*
+```
+If you’d like to run just a single Java test in the `ShuffleClientSuiteJ`, e.g., a test that with the name `testPushData`, you run the following command in the sbt console:
+
+```
+> testOnly *ShuffleClientSuiteJ -- *ShuffleClientSuiteJ.testPushData
+```
+
+If you’d like to run just a single Scala test in the `WorkerStatusTrackerSuite`, e.g., a test that incudes "handleHeartbeatResponse" in the name, you run the following command in the sbt console:
+
+```
+> testOnly *WorkerStatusTrackerSuite -- -z handleHeartbeatResponse
```

Comment:
can you use an example which contains space in the test name? IIRC, it's single/double quote sensitive

Reply:
updated, and replace all examples from the `client` project to the `master` project